roofit/roofitcore/inc/RooFit/Detail/ADContextManager.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Project: RooFit
+ * Authors:
+ *   Jonas Rembser, CERN 2023
+ *   Garima Singh, CERN 2023
+ *
+ * Copyright (c) 2023, CERN
+ *
+ * Redistribution and use in source and binary forms,
+ * with or without modification, are permitted according to the terms
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)
+ */
+
+#ifndef RooFit_Detail_ADContextManager_h
```

Comment:
I don't think the name `ADContextManager` makes sense. The code generation has nothing to do with AD a priori, and there are also other usecases for it. Generating standalone C++ code for your RooFit model for example, to use in other frameworks. Also, I think `ContextManager` is a bit verbose, and is not inline with existing RooFit code that follows the same pattern of visiting the computation graph. There is for example `ComplieContext` for compiling a graph for a given normalization set, or `RunContext` for evaluation in BatchMode.

I would therefore suggest calling this class `RooFit::Detail::CodeGenContext`. Does this make sense to you?

Reply:
Yep, I get it. However, I propose using `CodeSquashContext` instead.